src/commissaire_http/handlers/hosts.py

```diff
@@ -211,17 +211,21 @@ def delete_host(message, bus):
         bus.storage.delete(models.Host.new(address=address))
         # TODO: kick off service job to remove the host?
 
-        # Remove from a cluster
-        container = bus.storage.list(models.Clusters)
-        for cluster in container.clusters:
-            if address in cluster.hostset:
-                LOGGER.info('Removing host "{}" from cluster "{}"'.format(
-                    address, cluster.name))
-                cluster.hostset.pop(
-                    cluster.hostset.index(address))
-                bus.storage.save(cluster)
-                # A host can only be part of one cluster so break the loop
-                break
+        try:
+            # Remove from a cluster
+            container = bus.storage.list(models.Clusters)
+            for cluster in container.clusters:
+                if address in cluster.hostset:
+                    LOGGER.info('Removing host "{}" from cluster "{}"'.format(
+                        address, cluster.name))
+                    cluster.hostset.pop(
+                        cluster.hostset.index(address))
+                    bus.storage.save(cluster)
+                    # A host can only be part of one cluster so break the loop
+                    break
+        except _bus.RemoteProcedureCallError as error:
+            LOGGER.info('{} not part of a cluster.'.format(address))
+
```

Comment:
The `try...except` looks good here, so approving.

The following `except` clause assumes `RemoteProcedureCallError` means "address not found" when really it could mean any of a number of errors.  I'm guessing this is what inspired your `InstanceNotFoundError` suggestion https://github.com/projectatomic/commissaire-service/issues/41#issuecomment-268073051, which I'm considering.